truffleruby/src/main/ruby/core/truffle/cext.rb

```diff
@@ -1000,32 +1000,6 @@ def rb_String(value)
     String(value)
   end
 
-  def RARRAY_PTR(array)
-    ArrayPointer.new(array)
-  end
-
-  class ArrayPointer
-
-    attr_reader :array
-
-    def initialize(array)
-      @array = array
-    end
-
-    def size
-      array.size
-    end
-
-    def [](index)
-      array[index]
-    end
-
-    def []=(index, value)
-      array[index] = value
-    end
-
-  end
```

Comment:
Before, reads came in as a byte offset and writes as an array index. Now they're both array index so that just matches what an array does.